docs/yarn_classic.md

```diff
@@ -0,0 +1,178 @@
+it must contain the following lines:
+
+```ini
+yarn-offline-mirror <absolute path to the request output directory>
+yarn-offline-mirror-pruning false
+```
+
+It can be either directly written to or `yarn config` could be used:
+```bash
+$ yarn config set yarn-offline-mirror <absolute path to the request output directory>
+$ yarn config set yarn-offline-mirror-pruning false
+```
+
+In case when environment variables approach is preferred the following
+variables must be set:
+
+```bash
+YARN_YARN_OFFLINE_MIRROR=<absolute path to the request output directory>
+YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+```
+Cachi2 provides a helper that [generates these variables] and places them into a file.
+Sourcing this file is enough to set them.
+
+## Limitations and caveats
+
+### Yarn version specified anywhere in the package will be ignored by prefetch
+
+Unlike in the case of Yarn v3 Cachi2 will used whichever version is available system-wide
+on a system where a package is prefetched. In most practical cases this will default to
+the latest stable version of Yarn Classic (which is not under active
+development anymore).
+
+### Yarn Zero-Installs are not supported
+
+Yarn Classic's [Plug'n'Play] feature is not supported. Any package that uses
+it will be rejected. For further details please refer to [Yarn v3 documentation].
+
+### Handling of yarn-specific config files
+
+Yarn Classic allows a user to provide additional configuration via [.yarnrc]
+and [.npmrc].  **Cachi2 ignores these settings during prefetch phase**.
+However a `.yarnrc` could be used for setting up an offline mirror
+([Prerequisites for an offline build]).  These settings will be applied during
+a build phase.
+
+### Variables set during fetch phase
+
+The following variables are set for Yarn in the fetch phase:
+
+ * `COREPACK_ENABLE_DOWNLOAD_PROMPT` is set to "0" which prevents
+   Corepack from showing the URL when it needs to download software;
+ * `COREPACK_ENABLE_PROJECT_SPEC` is set to "0" which prevents
+   Corepack from checking if the package manager corresponds to the one
+   defined for the current project;
+ * `YARN_IGNORE_PATH` is set to "true" which ignores any Yarn version specified by a user and
+   uses Corepack's version instead;
+ * `YARN_IGNORE_SCRIPTS`: is set to "true" which prevents execution of any scripts defined in
+   `package.json` or in any dependency;
+ * `YARN_YARN_OFFLINE_MIRROR` is set to point to `deps/yarn-classic` which is relative to
+   output directory and will hold fetched dependencies;
+ * `YARN_YARN_OFFLINE_MIRROR_PRUNING` is set to "false" which prevents Yarn from attempting to
+   ensure that dependencies are up to date.
+
+Once fetch phase is completed Cachi2 will need to generate an
+[environment file] with variables pointing to the mirror
+and instructing Yarn not to prune it:
+
+```
+YARN_YARN_OFFLINE_MIRROR=<request output directory>
+YARN_YARN_OFFLINE_MIRROR_PRUNING=false
+```
```

Comment:
nitpick: this part is very similar to the one in `## Prerequisites for an offline build` and cachi2 `generate-env` command,  so I would merge them

Reply:
I would rather kept it here: it feels like a bit too much indirection for sucha small doc.